### PR TITLE
20354 money request participants selector skeleton fix

### PR DESCRIPTION
**Pull request by lukemorawski**
### Details

A quick fix to Skeleton issues on MoneyRequestParticipantsList components. Debouncing searching.

### Fixed Issues

$ https://github.com/Expensify/App/pull/32692#issuecomment-1888223436
PROPOSAL:

### Tests

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] make a money request
- [ ] observe the skeleton showing up when the screen shows up
- [ ] observe the skeleton showing up the searching for contact
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js [hunk 1]
```diff
@@ -79,14 +81,25 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({
...
 
+    const isOptionsDataReady = useMemo(() => ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(personalDetails), [personalDetails]);
+
+    useEffect(() => {
+        const interactionTask = InteractionManager.runAfterInteractions(() => {
+            setDidScreenTransitionEnd(true);
+        });
+
+        return interactionTask.cancel;
+    }, []);
```
Comment (rlinoz) [inline src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js:101]: Hey @lukemorawski can you help me understand why we need to check for the transition state?

**Comment by youssef-lr** [comment]
I think you're planning to reuse `BaseSelectionsList` in different other parent components right? Such as the SearchPage and NewChatPage, in this case, can we code this debouncing logic in the `BaseSelectionsList` itself instead of having to duplicate in each parent componenent?

**Comment by youssef-lr** [comment]
I'm still seeing a flicker, here it is compared to the search page

https://github.com/Expensify/App/assets/9680864/d042cfdb-068b-412b-8482-b7f7f4e863fb

**Comment by lukemorawski** [comment]
> I think you're planning to reuse `BaseSelectionsList` in different other parent components right? Such as the SearchPage and NewChatPage, in this case, can we code this debouncing logic in the `BaseSelectionsList` itself instead of having to duplicate in each parent componenent?

I actually wouldn't advice adding debouning to the list component. for simplicity and separation of concerns sake. The list is only to display things. Yes it has some convenient things added to it, but still they are "list related".
Let's say we added debouncing to `textInputValue` and suddenly we have a use-case where we actually don't need it because all the data is local and we just need to quickly filter them. We would then need to add yet another param to turn off debouncing. Also another param to manage to debouncing time. This exactly what happened to the previous list element (which we are removing right now) - it got bloated over time with features that we don't use 99% of time. By keeping debouncing outside of the scope of the list we keep things modular and easier to refactor in the future.

**Comment by lukemorawski** [comment]
> I'm still seeing a flicker, here it is compared to the search page
> 

OK, it's because the new `SelectionList` doesn't have a param that the old one had, which is a flag that was showing a loading indicator in input field. Instead I reverted to showing a skeleton while searching and that creates that unpleasant effect. I'll just add that old param to the new list component - this should solve the issue.

**Comment by brunovjk** [comment]
@lukemorawski Im working in this skeleton transition here #32665

Although you are still in the middle of refactoring, we need to [control ](https://github.com/Expensify/App/issues/31779#issuecomment-1829307739)the `getOptions` call. Like you did in [here](https://github.com/lukemorawski/ExpensifyApp/blob/6c2aa5113574cdce55289b7e6d1bc9fbd84a208b/src/pages/SearchPage/index.js#L72-L80).

Also,
> param that the old one had, which is a flag that was showing a loading indicator in input field

- I adjusted the 'SelectionList' props [#commit](https://github.com/Expensify/App/pull/32665/commits/301df39b60fc9f49b608b6932727f1eebd743c28), I think it works fine now, but any feedback is welcome. 
- We also need a `fill={theme.icon}` in the `MoneyRequestReferralProgramCTA.tsx`

We need to make the same adjustments to `NewChatPage` and `SearchPage`.
I went ahead and resolved this in my [PR](https://github.com/Expensify/App/pull/32665) as well. What do you think?

**Comment by lukemorawski** [comment]
> @lukemorawski Im working in this skeleton transition here #32665
> 
> Although you are still in the middle of refactoring, we need to [control ](https://github.com/Expensify/App/issues/31779#issuecomment-1829307739)the `getOptions` call. Like you did in [here](https://github.com/lukemorawski/ExpensifyApp/blob/6c2aa5113574cdce55289b7e6d1bc9fbd84a208b/src/pages/SearchPage/index.js#L72-L80).
> 
> Also,
> 
> > param that the old one had, which is a flag that was showing a loading indicator in input field
> 
> * I adjusted the 'SelectionList' props [#commit](https://github.com/Expensify/App/pull/32665/commits/301df39b60fc9f49b608b6932727f1eebd743c28), I think it works fine now, but any feedback is welcome.
> * We also need a `fill={theme.icon}` in the `MoneyRequestReferralProgramCTA.tsx`
> 
> I went ahead and resolved this in my [PR](https://github.com/Expensify/App/pull/32665) as well. What do you think?
> 
> We need to make the same adjustments to `NewChatPage` and `SearchPage`.

OK, I see we are basically working on mostly the same things here. There is no point of making two PRs with code changes addressing the same issues. I could close my PR if you could add some some extra things my PR have and your doesn't. That way we won't duplicate. If you're cool with that then here's a quick list:
1. debounce text input value. You can use the same way I used in this PR using `useDebouncedState` hook and `useEffect` to run `Report.searchInServer` on debounced value change.
2. applying all the same changes in `src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js` as it's basically the same component but used elsewhere.

**Comment by brunovjk** [comment]
> OK, I see we are basically working on mostly the same things here

Yes 😂

> I could close my PR if you could add some some extra things my PR have and your doesn't.

Absolutely, I will work on these changes immediately.

> And I could prepare another PR where fix the icon thing.

I completed the `fill` property in `MoneyRequestReferralProgramCTA` in my PR [#commit](https://github.com/Expensify/App/pull/32665/commits/6d15a87cc10a4f59e1b474272e563aa5968ba8c8), can I leave it? I believe we can create a `ReferralProgramCTA` to be used in all others, SerachPage and NewPage for example. I don't know if it already exists.

**Comment by brunovjk** [comment]
@lukemorawski,  One more thing, about https://github.com/Expensify/App/issues/20354, I believe we will have more work, I can see some places that use `OptionSelector` that I believe can be replaced by `SelectionList`, I don't know. But in any case, I'm at your disposal if you need help with the next steps of it.

**Comment by lukemorawski** [comment]
> @lukemorawski, One more thing, about #20354, I believe we will have more work, I can see some places that use `OptionSelector` that I believe can be replaced by `SelectionList`, I don't know. But in any case, I'm at your disposal if you need help with the next steps of it.

Thanks for that! I'm those refactorings one by one, so it's not a huge unreadable PR.

**Comment by lukemorawski** [comment]
> > OK, I see we are basically working on mostly the same things here
> 
> Yes 😂
> 
> > I could close my PR if you could add some some extra things my PR have and your doesn't.
> 
> Absolutely, I will work on these changes immediately.
> 
> > And I could prepare another PR where fix the icon thing.
> 
> I completed the `fill` property in `MoneyRequestReferralProgramCTA` in my PR [#commit](https://github.com/Expensify/App/pull/32665/commits/6d15a87cc10a4f59e1b474272e563aa5968ba8c8), can I leave it? I believe we can create a `ReferralProgramCTA` to be used in all others, SerachPage and NewPage for example. I don't know if it already exists.

Awesome. Leave your change please and I'll address the rest. I have already created one common `ReferralProgramCTA`. I'll just refer it in all those pages.

**Comment by lukemorawski** [comment]
Closing this PR because the work here has been largely double in https://github.com/Expensify/App/pull/32665. 
Please see this comment https://github.com/Expensify/App/pull/34443#issuecomment-1892308955

**Comment by brunovjk** [comment]
@lukemorawski The [PR](https://github.com/Expensify/App/pull/32665) is ready for Review.

I made the changes you mentioned:
> 1. debounce text input value. You can use the same way I used in this PR using useDebouncedState hook and useEffect to run Report.searchInServer on debounced value change.
> 2. applying all the same changes in src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js as it's basically the same component but used elsewhere.

Can you verify please? [#32665/files](https://github.com/Expensify/App/pull/32665/files)
Thank you :D

**Comment by brunovjk** [comment]
Commit changes requested by @lukemorawski here https://github.com/Expensify/App/pull/34443#issuecomment-1892308955 to resolve https://github.com/Expensify/App/pull/32692#issuecomment-1888223436.

**Pull request closed without merging**